bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -512,16 +507,8 @@ public void run() {
                 } catch (InterruptedException e) {
                     Thread.currentThread().interrupt();
                     LOG.info("ForceWrite thread interrupted");
-                    // close is idempotent
-                    if (!localRequests.isEmpty()) {
-                        ForceWriteRequest req = localRequests.get(localRequests.size() - 1);
-                        req.shouldClose = true;
-                        req.closeFileIfNecessary();
-                    }
```

Comment:
Won't this cause the journal to discard one flush?

Reply:
Here we catch InterruptedException, the InterruptedException is thrown by `forceWriteRequests.takeAll(localRequests)`.
So if it InterruptedException, that means we didn't read the data from the queue. So we don't need to handle the data.

Reply:
@merlimat, cc

Reply:
But it also interrupted the force write thread. We need to ensure the journal data is flushed into the disk, right?

Reply:
Maybe not, only the `forceWriteRequests.takeAll(localRequests)` throw the InterruptedException. If it throw InterruptedException, we didn't have data to flush.